diesel/src/pg/expression/functions.rs

```diff
@@ -1992,3 +1992,104 @@ define_sql_function! {
     /// ```
     fn jsonb_strip_nulls<E: JsonbOrNullableJsonb + SingleValue>(jsonb: E) -> E;
 }
+
+#[cfg(feature = "postgres_backend")]
+define_sql_function! {
+    /// Builds a JSON object out of a text array. The array must have an even number of members,
+    /// in which case they are taken as alternating key/value pairs. This function also has a form that
+    /// that takes keys and values as separate text array arguments.
+    /// See [jsonb_object_with_keys_and_values]
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// # include!("../../doctest_setup.rs");
+    /// #
+    /// # fn main() {
+    /// #     run_test().unwrap();
+    /// # }
+    /// #
+    /// # fn run_test() -> QueryResult<()> {
+    /// #     use diesel::dsl::jsonb_object;
+    /// #     use diesel::sql_types::{Array, Json, Nullable, Text};
+    /// #     use serde_json::Value;
+    /// #     let connection = &mut establish_connection();
+    /// let jsonb = diesel::select(jsonb_object::<Array<Text>,_>(vec!["hello","world"]))
+    ///                 .get_result::<Value>(connection)?;
+    /// let expected:Value = serde_json::json!({"hello":"world"});
+    /// assert_eq!(expected, jsonb);
+    ///
+    /// let jsonb = diesel::select(jsonb_object::<Array<Text>, _>(vec!["hello","world","John","Doe"]))
+    ///                 .get_result::<Value>(connection)?;
+    /// let expected:Value = serde_json::json!({"hello": "world","John": "Doe"});
+    /// assert_eq!(expected, jsonb);
+    ///
+    /// let jsonb = diesel::select(jsonb_object::<Array<Text>, _>(vec!["hello","world","John"]))
+    ///                 .get_result::<Value>(connection);
+    /// assert!(jsonb.is_err());
+    ///
+    /// let jsonb = diesel::select(jsonb_object::<Nullable<Array<Text>>, _>(None::<Vec<String>>))
+    ///                 .get_result::<Option<Value>>(connection);
+    /// assert!(jsonb.is_err());
```

Comment:
That does seem to be not correct, this should return `None` and no error.

Reply:
If I replace the current statement with this: 

```
let jsonb = diesel::select(jsonb_object::<Nullable<Array<Text>>, _>(None::<Vec<String>>))
           .get_result::<Option<Value>>(connection)?;
``` 

The test fails because of this error:

```
called `Result::unwrap()` on an `Err` value: DatabaseError(Unknown, "current transaction is aborted, commands ignored until end of transaction block")
```

I tried with different variations of the argument type, but couldn't get this error to resolve.

Reply:
That's likely caused by the error in the statement beforehand. Either get a new connection here or reorder the test cases to first have those without error and then these that return an error.